qutebrowser/mainwindow/prompt.py

```diff
@@ -645,9 +648,16 @@ def _directories_hide_show_model(self):
     @pyqtSlot(str)
     def _set_fileview_root(self, path, *, tabbed=False):
         """Set the root path for the file display."""
-        separators = os.sep
+        self._separators = os.sep
         if os.altsep is not None:
-            separators += os.altsep
+            self._separators += os.altsep
+
+        # If the path is not the same as the path with expanduser set a boolean
+        if os.path.expanduser(path) != path:
+            path = os.path.expanduser(path)
+            self._expands_user = True
+        elif not self._expands_user:
+            self._expands_user = False
```

Comment:
This section saves whether or not `os.path.expanduser` actually changes everything. If I don't do that and I type `~/` and then tab it will change to `/home/<user>/<dir>` which is not what you would expect.

Reply:
Please add a test for that scenario. Also, I don't understand the `elif` here - if it's set to false, set it to false? That seems redundant, no?